src/Native/Json.js

```diff
@@ -137,9 +137,11 @@ Elm.Native.Json.make = function(localRuntime) {
 			if (isObject) {
 				var keyValuePairs = List.Nil;
 				for (var key in value) {
-					var elmValue = decoder(value[key]);
-					var pair = Utils.Tuple2(key, elmValue);
-					keyValuePairs = List.Cons(pair, keyValuePairs);
+					if (value.hasOwnProperty(key)) {
```

Comment:
Why does this make this code better? Seems like it'll take some extra time and either (a) not protect us from anything because of how the Elm API is designed or (b) lead to weird results where we aren't allowed to grab fields that we want. Can you explain more about this?